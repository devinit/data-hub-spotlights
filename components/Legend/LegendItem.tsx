import React, { FunctionComponent } from 'react';

export interface LegendItemProps {
  className?: string;
  bgColor?: string;
  textColor?: string;
}

export const LegendItem: FunctionComponent<LegendItemProps> = ({ children, className, bgColor, textColor }) =>
  <span className={ className } data-testid="spotlight-legend-item">
    { children }
    <style jsx>{ `
      span {
        background-color: ${bgColor || 'inherit'};
        color: ${textColor || 'inherit'};
      }
    ` }</style>
  </span>;
