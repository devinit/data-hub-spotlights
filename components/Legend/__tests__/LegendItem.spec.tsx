/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { LegendItem, StyledLegendItem } from '../LegendItem';

describe('LegendItem', () => {
  test('renders correctly', () => {
    const renderer = TestRenderer.create(<StyledLegendItem />).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('renders with the specified class', () => {
    const { container } = render(<LegendItem className="legend-item" />);

    expect(container.firstChild).toHaveClass('legend-item');
  });

  test('renders with the specified children', () => {
    const { container } = render(
      <LegendItem>
        <span>30%</span>
      </LegendItem>
    );

    expect(container.firstChild).toMatchInlineSnapshot(`
      <span
        data-testid="spotlight-legend-item"
      >
        <span>
          30%
        </span>
      </span>
    `);
  });
});
