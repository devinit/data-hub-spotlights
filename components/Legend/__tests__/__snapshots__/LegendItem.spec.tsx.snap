// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LegendItem renders correctly 1`] = `
<span
  data-testid="spotlight-legend-item"
>
  <style
    jsx={true}
  >
    
      span {
        background-color: inherit;
        color: inherit;
      }
    
  </style>
</span>
`;

exports[`LegendItem renders with the specified children 1`] = `
<span
  data-testid="spotlight-legend-item"
>
  <span>
    30%
  </span>
  <style>
    
      span {
        background-color: inherit;
        color: inherit;
      }
    
  </style>
</span>
`;
